base/include/ArchiveSpaceManager.h

```diff
@@ -0,0 +1,71 @@
+#pragma once
+#include "Module.h"
+#include "AIPExceptions.h"
+
+class ArchiveSpaceManagerProps : public ModuleProps
+{
+public:
+	ArchiveSpaceManagerProps(uint32_t _lowerWaterMark, uint32_t _upperWaterMark, string _pathToWatch, int _samplingFreq)
```

Comment:
we should support another construcotr which just takes max_size_allowed and computes lower water mark from their as 10% less

Reply:
A constructor which takes max_size and computes lower water mark (10% less) is added.